Source/WebCore/rendering/style/RenderStyle.h

```diff
@@ -296,7 +296,7 @@ constexpr auto PseudoElementTypeBits = 5;
 DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(PseudoStyleCache);
 struct PseudoStyleCache {
     WTF_MAKE_STRUCT_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER(PseudoStyleCache);
-    Vector<std::unique_ptr<RenderStyle>, 4> styles;
+    HashMap<std::optional<Style::PseudoElementIdentifier>, std::unique_ptr<RenderStyle>> styles;
```

Comment:
Is std::.optional here just because we don't have traits for plain PseudoElementIdentifier? I don't think the key can be nullopt here. Maybe we could fix the traits and make the map bit more compact?

Reply:
Indeed, that was the exact reason, but it's not a great one.

I've added new traits for the not-optional variant, to avoid this.

Reply:
also we should really have generic traits for std::optional...